phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/GroupedAggregateRegionObserver.java

```diff
@@ -411,6 +487,115 @@ private UnorderedGroupByRegionScanner(final ObserverContext<RegionCoprocessorEnv
 
         @Override
         public boolean next(List<Cell> resultsToReturn) throws IOException {
+            if (firstScan && prevScanStartRowKey != null && prevScanIncludeStartRowKey != null) {
+                firstScan = false;
```

Comment:
Can't this logic when `firstScan` is true moved into the constructor and then we can remove this variable. One less state to maintain.

Reply:
If the region has moved, only then `firstScan` will be true while resuming the scan.

Reply:
But it is set to true where it is declared